src/scripts.js

```diff
@@ -4,85 +4,106 @@ import './css/style.scss';
-  let randomHistory = makeRandomDate(userRepo, userNowId, hydrationData);
-  historicalWeek.forEach(instance => instance.insertAdjacentHTML('afterBegin', `Week of ${randomHistory}`));
-  addInfoToUserSidebar(userNow, userRepo);
-  populateDailyData('hydration-today', hydrationRepo, userNow.id, today);
-  populateDailyData('activity-today', activityRepo, userNow.id, today);
-  populateDailyData('sleep-today', sleepRepo, userNow.id, today);
-  let winnerNow = makeWinnerID(activityRepo, userNow, today, userRepo);
+function catchData(src) {
+
+  let parent = findParentUrl(src);
+  return fetch(`https://fe-apps.herokuapp.com/api/v1/fitlit/1908/${parent}/${src}`)
+    .then(response => response.json())
+    .then(data => data[src]);
 }
 
-async function catchData(id) {
-  const response = await fetch(`https://fe-apps.herokuapp.com/api/v1/fitlit/1908/users/${id}`)
-  const data = await response.json();
-  const arrayData = await data[id];
-  return await arrayData
+function findParentUrl(src) {
+  if (src.includes('user')) {
+    return 'users'
+  } else if (src.includes('sleep')) {
+    return 'sleep'
+  } else if (src.includes('activity')) {
+    return 'activity'
+  } else if (src.includes('hydration')) {
+    return 'hydration'
+  }
 }
 
-async function makeUsers(array) {
-  const userData = await catchData('userData');
-  userData.forEach(function(dataItem) {
-    let user = new User(dataItem);
-    array.push(user);
+const data = {
+  users: catchData('userData'),
+  hydration: (catchData('hydrationData')),
+  activity: catchData('activityData'),
+  sleep: catchData('sleepData')
+};
+
+Promise.allSettled([data.users, data.sleep, data.activity, data.hydration])
+  .then(rawData => {
+    data.users = new UserRepo(rawData[0].value)
+    data.sleep = new SleepRepo(rawData[1].value)
+    data.activity = new ActivityRepo(rawData[2].value)
+    data.hydration = new HydrationRepo(rawData[3].value)
   })
+  // .then(result => console.log(result))
+  .then(startApp());
+
+function startApp() {
+console.log(data);
+//   var historicalWeek = document.querySelectorAll('.historicalWeek');
+  var userNowId = pickUser();
+  let userNow = getUserById(userNowId);
+  // let today = makeToday(userNow);
+//   let randomHistory = makeRandomDate(userRepo, userNowId, hydrationData);
+//   historicalWeek.forEach(instance => instance.insertAdjacentHTML('afterBegin', `Week of ${randomHistory}`));
+//   addInfoToUserSidebar(userNow, userRepo);
+//   ;
+//   populateDailyData('activity-today', activityRepo, userNow.id, today);
+//   populateDailyData('sleep-today', sleepRepo, userNow.id, today);
+//   let winnerNow = makeWinnerID(activityRepo, userNow, today, userRepo);
 }
 
 function pickUser() {
   return Math.floor(Math.random() * 50);
 }
 
-function getUserById(id, listRepo) {
-  return listRepo.getDataFromID(id);
+function getUserById(id) {
+  return data.users.then(data => data.find(user => user.id === id));
```

Comment:
I see what you mean about 'then', a new thing to attempt to wrap my mind around. Ill play around with it and get familiar with this new way to access data.